src/svc_auth_gss.c

```diff
@@ -544,6 +548,7 @@ _svcauth_gss(struct svc_req *req, bool *no_dispatch)
 
 		if (call_stat >= XPRT_DIED) {
 			rc = AUTH_FAILED;
+			(void)authgss_ctx_hash_del(gd);
 			goto gd_free;
 		}
 
```

Comment:
I wonder if this whole block is necessary.  At least established, I think.  Maybe just move the whole block above the reply?

Reply:
The block actually modifies gr.gr_major when it calls gss_get_name_attribute. And also sets the MSPAC flags in gd. So, I am not sure what effect is it going to have on the returned value. Maybe we can set established before sending the reply.